src/app/static/src/app/components/plots/MapLegend.vue

```diff
@@ -13,25 +13,41 @@
 <script lang="ts">
     import Vue from "vue";
     import {LControl} from 'vue2-leaflet';
+    import {colorFunctionFromName} from "../../store/filteredData/utils";
+
     export default Vue.extend({
         name: "MapLegend",
         props: {
-            "colorFunction": Function,
-            "max": Number,
-            "min": Number
+            "metadata": Object
         },
         components: {
             LControl
         },
         computed: {
             levels: function () {
-                const step = (this.max - this.min) / 5;
-                return [0, 1, 2, 3, 4, 5].map((i) => {
-                    const val = Math.round((this.min + (i * step)) * 100)/100;
-                    return {
-                        val, style: {background: this.colorFunction((val - this.min)/(this.max - this.min))}
-                    }
-                });
+                if (this.metadata) {
+                    const max = this.metadata.max;
+                    const min = this.metadata.min;
+                    const colorFunction = colorFunctionFromName(this.metadata.colour);
+                    const step = (max - min) / 5;
+
+                    //Allow levels to have 1 more decimal place than max value
```

Comment:
Not rounding at all didn't quite work, because you potentially get recurring decimals or floating point artefacts. This seems to work ok for the example we have!